webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,257 @@
+import React, {ReactNode} from 'react';
+import './ticketStyle.scss';
+
+import {Dispatch} from 'redux';
+
+import {Instance} from 'types/model';
+import DefaultAvatar from 'components/default_avatar/default_avatar';
+
+export type Props = {
+    href: string;
+    show: boolean;
+    connected: boolean;
+    ticketDetails?: TicketDetails;
+    connectedInstances: Instance[];
+    fetchIssueByKey: (issueKey: string, instanceID: string) => (dispatch: Dispatch, getState: any) => Promise<{
+        data?: TicketData;
+    }>;
+}
+
+export type State = {
+    isLoaded: boolean;
+    ticketId: string;
+    ticketDetails?: TicketDetails;
+};
+
+const isAssignedLabel = ' is assigned';
+const unAssignedLabel = 'Unassigned';
+const jiraTicketTitleMaxLength = 80;
+
+enum myStatus {
+    INDETERMINATE = 'indeterminate',
+    DONE = 'done',
+}
+
+const myStatusClasses: Record<string, string> = {
+    [myStatus.INDETERMINATE]: 'ticket-status--indeterminate',
+    [myStatus.DONE]: 'ticket-status--done',
+};
+
+export default class TicketPopover extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        const {ticketID} = this.getIssueKey();
+
+        this.state = {
+            isLoaded: false,
+            ticketId: ticketID,
+        };
+    }
+
+    getIssueKey = () => {
+        let ticketID = '';
+        let instanceID = '';
+
+        for (const instance of this.props.connectedInstances) {
+            instanceID = instance.instance_id;
+
+            try {
```

Comment:
We could save some time continuing when we know there is no instanceID.

```suggestion
            if(!this.props.href.includes(instanceID) {
              continue;
            }
            
            try { ... }
```

Reply:
@javaguirre I think that's what is happening currently. On line 59, if the instanceID does not match we will jump to the next iteration. Let me know if I am misinterpreting it!

Reply:
It's true, but this way you have less nested conditions and it's easier to follow, it's a nice to have though.